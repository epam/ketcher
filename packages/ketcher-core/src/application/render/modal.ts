// class Modal {
//   private modal: HTMLElement;
//   private modalContent: HTMLElement;

//   constructor() {
//     this.modal = document.createElement('div');
//     this.modal.className = 'modal';
//     this.modal.style.display = 'none';

//     this.modalContent = document.createElement('div');
//     this.modalContent.className = 'modal-content';

//     const closeModalButton = document.createElement('span');
//     closeModalButton.className = 'close';
//     closeModalButton.innerHTML = '&times;';

//     closeModalButton.addEventListener('click', () => {
//       this.close();
//     });

//     this.modalContent.appendChild(closeModalButton);

//     const modalText = document.createElement('p');
//     modalText.textContent = 'This is a modal window.';
//     this.modalContent.appendChild(modalText);

//     this.modal.appendChild(this.modalContent);

//     document.body.appendChild(this.modal);

//     window.addEventListener('click', (event) => {
//       if (event.target === this.modal) {
//         this.close();
//       }
//     });
//   }

//   open() {
//     this.modal.style.display = 'block';
//   }

//   close() {
//     this.modal.style.display = 'none';
//   }
// }

// export default Modal;
// import { createTheme } from '@mui/material/styles';

// const theme = createTheme();

export function openModal() {
  const modalRoot = document.createElement('div');
  modalRoot.id = 'modal-root';
  document.body.appendChild(modalRoot);

  const modalElement = document.createElement('div');
  modalRoot?.appendChild(modalElement);

  const handleClose = () => {
    modalRoot?.removeChild(modalElement);
  };

  const modal = document.createElement('div');
  modal.className = 'modal';

  const modalContent = document.createElement('div');
  modalContent.className = 'modal-content';

  modalContent.innerHTML = `
      <div style="padding: 12px; min-height: calc(100% - 52px); border-bottom: 1px solid #e1e5ea; display: flex;">
        <div style="">
          <div>Name 1</div>
          <div style="border: 1px solid #e1e5ea; border-radius: 8px; width: 230px; height: 230px;"></div>
          <div>attachment points 1</div>
        </div>
        <div style="">
          <div>Name 2</div>
          <div style="border: 1px solid #e1e5ea; border-radius: 8px; width: 230px; height: 230px;"></div>
          <div>attachment points 2</div>
        </div>
      </div>
      <footer style="display: flex; justify-content: space-between; max-width: 100%; padding: 16px 12px 12px; gap: 8px;">
        <button id="closeButton" style="display: inline-flex; justify-content: center; align-items: center; outline: none; min-width: 70px; line-height: 14px; height: 24px; border-radius: 4px; font-size: 12px; color: #585858; border: 1px solid #585858; background-color: #ffffff; text-transform: none; font-weight: 400;">Cancel</button>
        <input type="button" disabled style="display: inline-flex; justify-content: center; align-items: center; outline: none; min-width: 70px; line-height: 14px; height: 24px; border-radius: 4px; font-size: 12px; color: #ffffff; background-color: #167782; border: 1px solid transparent;" title="Bond will be added to canvas" value="Connect">
      </footer>
    `;

  modal.appendChild(modalContent);

  const modalStyle = `
    display: flex;
    align-items: center;
    justify-content: center;
    position: absolute;
    left: 0;
    top: 0;
    bottom: 0;
    right: 0;
    z-index: 40;
    background-image: url(data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%2210%22%20height%3D%2210%22%3E%3Cpath%20d%3D%22M0%2010L10%200zm12-2l-4%204zM-2%202l4-4z%22%20stroke%3D%22%23555%22%20stroke-width%3D%222%22%20stroke-opacity%3D%22.02%22%2F%3E%3C%2Fsvg%3E);
    background-color: rgba(94, 94, 94, 0.2);
    background: none;
    filter: progid:DXImageTransform.Microsoft.gradient(startColorstr='#335e5e5e', endColorstr='#335e5e5e');
    `;

  const modalContentStyle = `
    min-width: 500px;
    min-height: 500px;
    outline: 0;
    position: relative;
    vertical-align: middle;
    display: flex;
    flex-direction: column;
    text-align: left;
    background-color: #ffffff;
    border-radius: 8px;
    background-clip: padding-box;
    font-family: Inter, FreeSans, Arimo, 'Droid Sans', Helvetica, 'Helvetica Neue', Arial, sans-serif;
    color: #585858;
    opacity: 1;
    transform: scale(1);
    transition: transform 0.3s, opacity 0.3s;
    box-shadow: 0 5px 20px rgba(103, 104, 132, 0.25);
    background-color: #ffffff;
    `;

  modal.style.cssText = modalStyle;
  modalContent.style.cssText = modalContentStyle;

  modalElement.appendChild(modal);

  modalElement.style.display = 'block';

  // Add event listener for close button
  const closeButton = document.getElementById('closeButton');
  closeButton?.addEventListener('click', handleClose);
}
