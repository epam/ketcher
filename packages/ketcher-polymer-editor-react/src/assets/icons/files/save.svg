<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M16.999 2.99998H4.99902C3.88902 2.99998 2.99902 3.89998 2.99902 4.99998V19C2.99902 20.1 3.88902 21 4.99902 21H18.999C20.099 21 20.999 20.1 20.999 19V6.99998L16.999 2.99998ZM18.999 19H4.99902V4.99998H16.169L18.999 7.82998V19ZM6.49902 6.49998H14.499V9.49998H6.49902V6.49998Z" fill="#005662"/>
<path d="M13.999 15C13.999 16.1045 13.1036 17 11.999 17C10.8945 17 9.99902 16.1045 9.99902 15C9.99902 13.8954 10.8945 13 11.999 13C13.1036 13 13.999 13.8954 13.999 15Z" fill="#005662"/>
</svg>
