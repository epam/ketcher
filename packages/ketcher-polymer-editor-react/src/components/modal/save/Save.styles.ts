import styled from '@emotion/styled';
import { DropDown } from 'components/shared/dropDown';

export const Form = styled.form({
  display: 'flex',
  flexDirection: 'column',
  height: '300px',
});

export const Row = styled.div({
  display: 'flex',
  justifyContent: 'space-between',
  marginBottom: '16px',
});

export const StyledDropdown = styled(DropDown)(({ theme }) => ({
  width: 'calc(50% - 6px)',
  height: '28px',

  '& .MuiOutlinedInput-root:hover': {
    border: `1px solid ${theme.ketcher.color.input.border.hover}`,
  },

  '& .MuiOutlinedInput-root': {
    border: `1px solid ${theme.ketcher.color.input.border.regular}`,
    backgroundColor: theme.ketcher.color.background.primary,
    color: theme.ketcher.color.text.primary,
    fontFamily: theme.ketcher.font.family.inter,
  },
}));

export const stylesForExpanded = {
  border: 'none',
};

export const Loader = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background: #fff;
`;
