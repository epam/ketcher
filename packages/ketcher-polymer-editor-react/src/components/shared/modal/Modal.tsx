import {
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  IconButton
} from '@mui/material'
import React, { useMemo } from 'react'
import { useTheme } from '@emotion/react'
import styled from '@emotion/styled'
import { Icon } from 'ketcher-react'
import { scrollbarThin } from 'theming/mixins'

interface ModalProps {
  children: JSX.Element | Array<JSX.Element>
  title: string
  isOpen: boolean
  showCloseButton?: boolean
  onClose: () => void
  className?: string
}

const Header = styled(DialogTitle)(({ theme }) => ({
  margin: '12px 11px 12px 16px',
  padding: 0,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-between',
  fontFamily: `${theme.ketcher.font.family.inter}`,
  fontSize: `${theme.ketcher.font.size.medium}`,
  fontWeight: `${theme.ketcher.font.weight.regular}`,
  textTransform: 'uppercase'
}))

const Title = styled.div({
  marginRight: '10px'
})

const Content = styled(DialogContent)(({ theme }) => ({
  margin: '0 16px 16px',
  padding: 0,

  ...scrollbarThin(theme)
}))

const Footer = styled(DialogActions)({
  margin: '0 16px 16px',
  padding: 0
})

type ModalSubcomponent = 'Content' | 'Footer'

export const Modal = ({
  children,
  title,
  isOpen,
  showCloseButton = true,
  onClose,
  className
}: ModalProps) => {
  const theme = useTheme()

  const paperProps = useMemo(
    () => ({
      style: {
        minWidth: '20vw',
        background: theme.ketcher.color.background.canvas,
        borderRadius: '8px',
        color: theme.ketcher.color.text.primary
      }
    }),
    [theme.ketcher.color.text.primary, theme.ketcher.color.background.canvas]
  )

  const backdropProps = useMemo(
    () => ({
      style: {
        background: theme.ketcher.color.background.overlay,
        opacity: 0.4
      }
    }),
    [theme.ketcher.color.background.overlay]
  )

  const subcomponents: Record<ModalSubcomponent, JSX.Element | null> = {
    Content: null,
    Footer: null
  }

  React.Children.forEach(children, (child) => {
    if (child.type === Content) {
      subcomponents.Content = child
    } else if (child.type === Footer) {
      subcomponents.Footer = child
    }
  })

  return (
    <Dialog
      BackdropProps={backdropProps}
      PaperProps={paperProps}
      open={isOpen}
      maxWidth="md"
      onClose={onClose}
      disableEscapeKeyDown={!showCloseButton}
      className={className}
    >
      <Header>
        <Title>{title}</Title>
        {showCloseButton && (
          <IconButton title={'Close window'} onClick={onClose}>
            <Icon name={'close'} />
          </IconButton>
        )}
      </Header>

      {subcomponents.Content}

      {subcomponents.Footer}
    </Dialog>
  )
}

Modal.Content = Content
Modal.Footer = Footer
