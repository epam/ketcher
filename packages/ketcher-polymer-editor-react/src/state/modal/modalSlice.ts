/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/
import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { RootState } from 'state';

interface ModalState {
  name: string | null;
  isOpen: boolean;
}

const initialState: ModalState = {
  name: null,
  isOpen: false,
};

export const modalSlice = createSlice({
  name: 'modal',
  initialState,
  reducers: {
    openModal: (state, action: PayloadAction<string>) => {
      state.name = action.payload;
      state.isOpen = true;
    },
    closeModal: (state) => {
      state.name = null;
      state.isOpen = false;
    },
  },
});

export const { openModal, closeModal } = modalSlice.actions;

export const selectModalName = (state: RootState): string | null =>
  state.modal.name;
export const selectModalIsOpen = (state: RootState): boolean =>
  state.modal.isOpen;

export const modalReducer = modalSlice.reducer;
