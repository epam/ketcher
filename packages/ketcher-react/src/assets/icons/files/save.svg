<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none">
<path d="M17.8333 18.2083H18.2083V17.8333V8.525V8.36967L18.0985 8.25983L15.7402 5.9015L15.6303 5.79167H15.475H6.16667H5.79167V6.16667V17.8333V18.2083H6.16667H17.8333ZM6.16667 4.875H16.0113L19.125 7.98866V17.8333C19.125 18.5429 18.5429 19.125 17.8333 19.125H6.16667C5.44999 19.125 4.875 18.5441 4.875 17.8333V6.16667C4.875 5.45588 5.44999 4.875 6.16667 4.875ZM7.79167 7.79167H13.7083V9.54167H7.79167V7.79167ZM13.2917 14.5C13.2917 15.2134 12.7134 15.7917 12 15.7917C11.2866 15.7917 10.7083 15.2134 10.7083 14.5C10.7083 13.7866 11.2866 13.2083 12 13.2083C12.7134 13.2083 13.2917 13.7866 13.2917 14.5Z" fill="currentColor" stroke="currentColor" stroke-width="0.75"/>
</svg>