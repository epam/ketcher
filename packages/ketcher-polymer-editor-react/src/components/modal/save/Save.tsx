/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/

import { useEffect, useState } from 'react'
import styled from '@emotion/styled'

import { Modal } from 'components/shared/modal'
import { DropDown } from 'components/shared/dropDown'
import { Option } from 'components/shared/dropDown/dropDown'
import { TextField } from 'components/shared/textEditor'
import { TextInputField } from 'components/shared/textInputField'
import { SaveButton } from 'components/modal/save/saveButton'
import { getPropertiesByFormat, SupportedFormats } from 'helpers/formats'
import { ActionButton } from 'components/shared/actionButton'
import { Icon } from 'ketcher-react'

interface Props {
  onClose: () => void
  isModalOpen: boolean
}

const options: Array<Option> = [
  { id: 'mol', label: 'MDL Molfile V3000' },
  { id: 'helm', label: 'HELM' }
]

const Form = styled.form({
  display: 'flex',
  flexDirection: 'column',
  height: '100%'
})

const Row = styled.div({
  display: 'flex',
  justifyContent: 'space-between',
  marginBottom: '16px'
})

const Label = styled.label(({ theme }) => ({
  marginRight: '8px',
  color: theme.ketcher.color.text.secondary
}))

const StyledDropdown = styled(DropDown)(({ theme }) => ({
  '& .MuiOutlinedInput-root': {
    border: 'none',
    backgroundColor: theme.ketcher.color.background.primary,
    color: theme.ketcher.color.text.primary,
    fontFamily: theme.ketcher.font.family.inter
  }
}))

const stylesForExpanded = {
  border: 'none'
}

const StyledModal = styled(Modal)({
  '& .MuiPaper-root': {
    width: '520px',
    height: '358px'
  },

  '& .MuiDialogContent-root': {
    overflow: 'hidden'
  }
})

const ErrorsButton = styled(ActionButton)(({ theme }) => ({
  backgroundColor: theme.ketcher.color.background.canvas,
  color: theme.ketcher.color.text.error,

  '&:hover': {
    backgroundColor: 'initial'
  }
}))

const structExample = {
  mol: `
    -INDIGO-02102212382D
  
    0  0  0  0  0  0  0  0  0  0  0 V3000
  M  V30 BEGIN CTAB
  M  V30 COUNTS 11 10 0 0 0
  M  V30 BEGIN ATOM
  M  V30 1 C 2.75 -5.225 0.0 0
  M  V30 2 C 3.61603 -5.725 0.0 0
  M  V30 3 C 4.48205 -5.225 0.0 0
  M  V30 4 C 5.34808 -5.725 0.0 0
  M  V30 5 C 6.2141 -5.225 0.0 0
  M  V30 6 C 7.08013 -5.725 0.0 0
  M  V30 7 C 7.94615 -5.225 0.0 0
  M  V30 8 C 8.81218 -5.725 0.0 0
  M  V30 9 C 9.6782 -5.225 0.0 0
  M  V30 10 C 10.5442 -5.725 0.0 0
  M  V30 11 C 11.4103 -5.225 0.0 0
  M  V30 END ATOM
  M  V30 BEGIN BOND
  M  V30 1 1 1 2
  M  V30 2 1 2 3
  M  V30 3 1 3 4
  M  V30 4 1 4 5
  M  V30 5 1 5 6
  M  V30 6 1 6 7
  M  V30 7 1 7 8
  M  V30 8 1 8 9
  M  V30 9 1 9 10
  M  V30 10 1 10 11
  M  V30 END BOND
  M  V30 END CTAB
  M  END
  `,
  helm: `PEPTIDE1{A.C.D.F.G.H.K.A.C.D}$PEPTIDE1,PEPTIDE1,3:R3-7:R3`
} // TODO remove when canvas and get struct method are ready

export const Save = ({ onClose, isModalOpen }: Props): JSX.Element => {
  const [currentFileFormat, setCurrentFileFormat] =
    useState<SupportedFormats>('mol')
  const [currentFileName, setCurrentFileName] = useState('ketcher')
  const [struct, setStruct] = useState('')
  const [errors, setErrors] = useState('')

  const handleSelectChange = (value) => {
    setCurrentFileFormat(value)
  }

  const handleInputChange = (value) => {
    setCurrentFileName(value)
  }

  const handleSave = () => {
    console.log('Saved', structExample)
  }

  const handleErrorsClick = () => {
    console.log('errors...')
  }

  useEffect(() => {
    console.log('Getting and setting struct here...') // get / convert struct and errors
    setStruct(structExample[currentFileFormat])

    if (currentFileFormat === 'mol') {
      // just an example
      setErrors('some error')
    } else {
      setErrors('')
    }
  }, [currentFileFormat])

  return (
    <StyledModal title="save structure" isOpen={isModalOpen} onClose={onClose}>
      <Modal.Content>
        <Form onSubmit={handleSave} id="save">
          <Row>
            <div>
              <TextInputField
                value={currentFileName}
                id="filename"
                onChange={handleInputChange}
                label="File name:"
              />
            </div>
            <div style={{ lineHeight: '24px' }}>
              <Label htmlFor="fileformat">File format:</Label>
              <StyledDropdown
                options={options}
                currentSelection={currentFileFormat}
                selectionHandler={handleSelectChange}
                customStylesForExpanded={stylesForExpanded}
              />
            </div>
          </Row>
          <TextField struct={struct} readonly selectOnInit />
        </Form>
      </Modal.Content>

      <Modal.Footer>
        {errors && (
          <ErrorsButton label="Errors" clickHandler={handleErrorsClick}>
            <Icon name="error" />
            Errors
          </ErrorsButton>
        )}

        <SaveButton
          label="Save as file"
          data={struct}
          type={getPropertiesByFormat(currentFileFormat).mime}
          onSave={handleSave}
          filename={
            currentFileName +
            getPropertiesByFormat(currentFileFormat).extensions[0]
          }
          disabled={!currentFileName}
        />
      </Modal.Footer>
    </StyledModal>
  )
}
