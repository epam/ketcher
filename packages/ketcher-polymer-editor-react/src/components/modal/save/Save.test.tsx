/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/

import { act, fireEvent, render, screen } from '@testing-library/react';
import { Save } from 'components/modal/save';
import userEvent from '@testing-library/user-event';

const mockOnClose = jest.fn();

const mockProps = {
  onClose: mockOnClose,
  isModalOpen: true,
};

describe('Save modal', () => {
  it('renders correctly', () => {
    const view = render(withThemeProvider(<Save {...mockProps} />));

    const filenameInput = screen.getByRole('textbox', {
      name: 'File name:',
    });
    const fileFormatInput = screen.getByRole('button', {
      name: 'MDL Molfile V3000',
    });

    expect(view).toMatchSnapshot();
    expect(filenameInput).toBeVisible();
    expect(filenameInput).toHaveValue('ketcher');
    expect(fileFormatInput).toBeVisible();
  });

  it.skip('renders dropdown options correctly', () => {
    render(withThemeProvider(<Save {...mockProps} />));

    const fileFormat = screen.getByRole('button', {
      name: 'MDL Molfile V3000',
    });

    fireEvent.click(fileFormat);
    const fileFormatDropdown = screen.getByTestId('dropdown-select');
    const option1 = screen.getByRole('option', { name: 'MDL Molfile V3000' });
    const option2 = screen.getByRole('option', { name: 'HELM' });

    expect(fileFormatDropdown).toBeVisible();
    expect(option1).toBeVisible();
    expect(option2).toBeVisible();
  });

  it.skip('renders buttons correctly', () => {
    render(withThemeProvider(<Save {...mockProps} />));

    const saveButton = screen.getByRole('button', { name: 'Save as file' });
    const filenameInput = screen.getByRole('textbox', {
      name: 'File name:',
    });

    // eslint-disable-next-line testing-library/no-unnecessary-act
    act(() => {
      userEvent.clear(filenameInput);
    });

    expect(saveButton).toBeDisabled();
  });
});
