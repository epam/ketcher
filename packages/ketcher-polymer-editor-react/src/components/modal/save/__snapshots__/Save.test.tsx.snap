// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Save modal renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiModal-root MuiDialog-root css-1hqci05-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 0.4; background: rgb(0, 86, 98); webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby=":r0:"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthMd css-12rl710-MuiPaper-root-MuiDialog-paper"
          role="dialog"
          style="min-width: 20vw; background: rgb(242, 242, 242); border-radius: 8px; color: rgb(51, 51, 51);"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-1x5ms5d-MuiTypography-root-MuiDialogTitle-root"
            id=":r0:"
          >
            <div
              class="css-lvyu5j"
            >
              save structure
            </div>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              title="Close window"
              type="button"
            >
              <icon-close.svg
                fill="#333333"
                role="img"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </h2>
          <div
            class="MuiDialogContent-root css-1n1tez8-MuiDialogContent-root"
          >
            <form
              class="css-1g4yje1"
              id="save"
            >
              <div
                class="css-1abudc7"
              >
                <div>
                  <label
                    class="css-14c8znx"
                    for="filename"
                  >
                    File name:
                  </label>
                  <input
                    class="css-yab8h5"
                    id="filename"
                    type="text"
                    value="ketcher"
                  />
                </div>
                <div
                  style="line-height: 24px;"
                >
                  <label
                    class="css-14c8znx"
                    for="fileformat"
                  >
                    File format:
                  </label>
                  <div
                    class="MuiFormControl-root css-dk6fk8-MuiFormControl-root"
                  >
                    <div
                      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-objd0-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                    >
                      <div
                        aria-expanded="false"
                        aria-haspopup="listbox"
                        class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                        role="button"
                        tabindex="0"
                      >
                        <span>
                          MDL Molfile V3000
                        </span>
                      </div>
                      <input
                        aria-hidden="true"
                        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                        tabindex="-1"
                        value="mol"
                      />
                      <icon-chevron.svg
                        classname="MuiSelect-icon MuiSelect-iconOutlined css-1bn8sss-MuiSelect-icon"
                        fill="#333333"
                        role="img"
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-hdw1oc"
                        >
                          <span
                            class="notranslate"
                          >
                            â€‹
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </div>
              <textarea
                class="css-17sxpbp"
                readonly=""
              >
                
    -INDIGO-02102212382D
  
    0  0  0  0  0  0  0  0  0  0  0 V3000
  M  V30 BEGIN CTAB
  M  V30 COUNTS 11 10 0 0 0
  M  V30 BEGIN ATOM
  M  V30 1 C 2.75 -5.225 0.0 0
  M  V30 2 C 3.61603 -5.725 0.0 0
  M  V30 3 C 4.48205 -5.225 0.0 0
  M  V30 4 C 5.34808 -5.725 0.0 0
  M  V30 5 C 6.2141 -5.225 0.0 0
  M  V30 6 C 7.08013 -5.725 0.0 0
  M  V30 7 C 7.94615 -5.225 0.0 0
  M  V30 8 C 8.81218 -5.725 0.0 0
  M  V30 9 C 9.6782 -5.225 0.0 0
  M  V30 10 C 10.5442 -5.725 0.0 0
  M  V30 11 C 11.4103 -5.225 0.0 0
  M  V30 END ATOM
  M  V30 BEGIN BOND
  M  V30 1 1 1 2
  M  V30 2 1 2 3
  M  V30 3 1 3 4
  M  V30 4 1 4 5
  M  V30 5 1 5 6
  M  V30 6 1 6 7
  M  V30 7 1 7 8
  M  V30 8 1 8 9
  M  V30 9 1 9 10
  M  V30 10 1 10 11
  M  V30 END BOND
  M  V30 END CTAB
  M  END
  
              </textarea>
            </form>
          </div>
          <div
            class="MuiDialogActions-root MuiDialogActions-spacing css-1codpvf-MuiDialogActions-root"
          >
            <button
              class="MuiButtonBase-root css-1uk7g1a-MuiButtonBase-root"
              role="button"
              tabindex="0"
              title="Errors"
              type="button"
            >
              Errors
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root css-1u2dyik-MuiButtonBase-root"
              role="button"
              tabindex="0"
              title="Save as file"
              type="submit"
            >
              Save as file
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
